common/src/main/java/io/netty/util/internal/SWARUtil.java

```diff
@@ -29,14 +29,31 @@ public static long compilePattern(byte byteToFind) {
 
     /**
      * Applies a compiled pattern to given word.
-     * Returns the index of the first occurrence of byte that specified in the pattern.
+     * Returns a word where each byte that matches the pattern has the highest bit set.
+     *
+     * @param word    the word to apply the pattern to
+     * @param pattern the pattern to apply
+     * @return a word where each byte that matches the pattern has the highest bit set
      */
-    public static int firstAnyPattern(long word, long pattern, boolean leading) {
+    public static long applyPattern(final long word, final long pattern) {
         long input = word ^ pattern;
         long tmp = (input & 0x7F7F7F7F7F7F7F7FL) + 0x7F7F7F7F7F7F7F7FL;
-        tmp = ~(tmp | input | 0x7F7F7F7F7F7F7F7FL);
-        final int binaryPosition = leading? Long.numberOfLeadingZeros(tmp) : Long.numberOfTrailingZeros(tmp);
-        return binaryPosition >>> 3;
+        return ~(tmp | input | 0x7F7F7F7F7F7F7F7FL);
+    }
+
+    /**
+     * Returns the index of the first occurrence of byte that specificied in the pattern.
+     * If no pattern is found, returns 8.
+     *
+     * @param word     the return value of {@link #applyPattern(long, long)}
+     * @param isNative if ture, if given word is big endian
```

Comment:
```suggestion
     * @param isNative if true, if given word is big endian
```

Reply:
Also, let's call the parameter `isBigEndian` because `isNative` is a context-specific variable of `firstIndexOf()`.

Reply:
done :D